File: back/rabbitmq_test.py
import pika
import time
import json

# RabbitMQ connection parameters
RABBITMQ_HOST = 'localhost'
RABBITMQ_PORT = 5672
RABBITMQ_USER = 'admin'
RABBITMQ_PASS = 'admin'
RABBITMQ_QUEUE = 'test_queue'

def send_message():
    """Send a message to RabbitMQ"""
    try:
        # Create connection
        connection = pika.BlockingConnection(
            pika.ConnectionParameters(
                host=RABBITMQ_HOST,
                port=RABBITMQ_PORT,
                credentials=pika.PlainCredentials(RABBITMQ_USER, RABBITMQ_PASS)
            )
        )
        channel = connection.channel()
        
        # Declare queue
        channel.queue_declare(queue=RABBITMQ_QUEUE, durable=True)
        
        # Send message
        message = {
            'timestamp': time.time(),
            'message': 'Hello from Python!',
            'type': 'test'
        }
        
        channel.basic_publish(
            exchange='',
            routing_key=RABBITMQ_QUEUE,
            body=json.dumps(message),
            properties=pika.BasicProperties(
                delivery_mode=2,  # make message persistent
            )
        )
        
        print(f" [✅] Sent message: {message}")
        connection.close()
        
    except Exception as e:
        print(f" [❌] Error sending message: {e}")

def receive_message():
    """Receive messages from RabbitMQ"""
    try:
        # Create connection
        connection = pika.BlockingConnection(
            pika.ConnectionParameters(
                host=RABBITMQ_HOST,
                port=RABBITMQ_PORT,
                credentials=pika.PlainCredentials(RABBITMQ_USER, RABBITMQ_PASS)
            )
        )
        channel = connection.channel()
        
        # Declare queue
        channel.queue_declare(queue=RABBITMQ_QUEUE, durable=True)
        
        def callback(ch, method, properties, body):
            """Callback function for received messages"""
            message = json.loads(body)
            print(f" [📥] Received message: {message}")
            ch.basic_ack(delivery_tag=method.delivery_tag)
        
        # Set up consumer
        channel.basic_consume(
            queue=RABBITMQ_QUEUE,
            on_message_callback=callback,
            auto_ack=False
        )
        
        print(' [*] Waiting for messages. To exit press CTRL+C')
        channel.start_consuming()
        
    except Exception as e:
        print(f" [❌] Error receiving message: {e}")

if __name__ == '__main__':
    print("🐰 RabbitMQ Python Test")
    print("Choose option:")
    print("1. Send message")
    print("2. Receive messages")
    
    choice = input("Enter choice (1 or 2): ")
    
    if choice == '1':
        send_message()
    elif choice == '2':
        receive_message()
    else:
        print("Invalid choice")

File: Downloads/Ecommerce_Microservices-fullIntegration/Ecommerce_Microservices-fullIntegration/back/rabbitmq_test.js
const amqp = require("amqplib");

// RabbitMQ connection parameters
const RABBITMQ_URL = "amqp://admin:admin@localhost:5672";
const RABBITMQ_QUEUE = "test_queue";

async function sendMessage() {
  try {
    // Create connection
    const connection = await amqp.connect(RABBITMQ_URL);
    const channel = await connection.createChannel();

    // Declare queue
    await channel.assertQueue(RABBITMQ_QUEUE, { durable: true });

    // Send message
    const message = {
      timestamp: new Date().toISOString(),
      message: "Hello from JavaScript!",
      type: "test"
    };

    channel.sendToQueue(RABBITMQ_QUEUE, Buffer.from(JSON.stringify(message)), {
      persistent: true
    });

    console.log(" [✅] Sent message:", message);

    await channel.close();
    await connection.close();
  } catch (error) {
    console.error(" [❌] Error sending message:", error.message);
  }
}

async function receiveMessage() {
  try {
    // Create connection
    const connection = await amqp.connect(RABBITMQ_URL);
    const channel = await connection.createChannel();

    // Declare queue
    await channel.assertQueue(RABBITMQ_QUEUE, { durable: true });

    console.log(" [*] Waiting for messages. To exit press CTRL+C");

    // Set up consumer
    await channel.consume(RABBITMQ_QUEUE, msg => {
      if (msg) {
        const message = JSON.parse(msg.content.toString());
        console.log(" [📥] Received message:", message);
        channel.ack(msg);
      }
    });
  } catch (error) {
    console.error(" [❌] Error receiving message:", error.message);
  }
}

// Main function
async function main() {
  console.log("🐰 RabbitMQ JavaScript Test");
  console.log("Choose option:");
  console.log("1. Send message");
  console.log("2. Receive messages");

  const readline = require("readline");
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  rl.question("Enter choice (1 or 2): ", async choice => {
    if (choice === "1") {
      await sendMessage();
    } else if (choice === "2") {
      await receiveMessage();
    } else {
      console.log("Invalid choice");
    }
    rl.close();
  });
}

main().catch(console.error);
